packages/core/src/execution-engine/partial-execution-utils/directed-graph.ts

```diff
@@ -50,6 +50,25 @@ export class DirectedGraph {
 		return new Map(this.nodes.entries());
 	}
 
+	/**
+	 * Returns a set of nodes whose names match the provided array of names.
+	 *
+	 * Only nodes that exist in the graph will be included in the result.
+	 */
+	getNodesByNames(names: string[]) {
```

Comment:
we seem to keep adding very similar code between `Workflow` and `DirectedGraph`. Perhaps we should consider consolidating the two into a single class.

I'm certain that frontend code could also benefit from all the additional (and well documented) functionality that `DirectedGraph` implements.

Reply:
I can take a stab at replacing the graph logic in workflow with the directed graph. I think it's beneficial to have the graph stuff in its own class to keep the workflow class smaller and more related to n8n specific stuff.